src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -284,7 +284,17 @@ protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder
             {
                 diagnostics.Add(ErrorCode.WRN_NullAsNonNullable, parameterSyntax.Default.Value.Location);
             }
-
+            if (this.IsNullChecked)
+            {
+                if (convertedExpression.ConstantValue?.IsNull == true)
+                {
+                    diagnostics.Add(ErrorCode.WRN_NullCheckedHasDefaultNull, this.Locations.FirstOrNone(), new object[] { this });
```

Comment:
>this.Locations.FirstOrNone() [](start = 77, length = 28)

Should this be reported on the `null` value rather than the parameter, as in the warning above?